pkg/resources/validation/podsecurityadmissionconfigurationtemplate/podsecurityact.go

```diff
@@ -0,0 +1,295 @@
+		resp.Result = &metav1.Status{
+			Status:  "Failure",
+			Message: fmt.Sprintf("Cannot delete built-in template '%s'", oldTemplate.Name),
+			Reason:  metav1.StatusReasonForbidden,
+			Code:    http.StatusForbidden,
+		}
+		resp.Allowed = false
+		return
+	}
+
+	// we can't allow templates to be deleted if they are being used by active clusters. Depending on the distro,
+	// the template reference could be stored on the v1.Cluster or v3.Cluster.
+	mgmtClusters, err := v.ManagementClusterCache.GetByIndex(ByPodSecurityAdmissionConfigurationName, oldTemplate.Name)
+	if reportDeletionError(len(mgmtClusters), err, "management", oldTemplate.Name, resp) {
+		return
+	}
+
+	provClusters, err := v.provisioningClusterCache.GetByIndex(ByPodSecurityAdmissionConfigurationName, oldTemplate.Name)
+	if reportDeletionError(len(provClusters), err, "provisioning", oldTemplate.Name, resp) {
+		return
+	}
+
+}
+
+func (v *Validator) ValidateConfiguration(configurationTemplate *mgmtv3.PodSecurityAdmissionConfigurationTemplate, resp *admissionv1.AdmissionResponse) {
+	configuration := configurationTemplate.Configuration
+	// validate defaults
+	if reportError(validateLevel(field.NewPath("defaults", "enforce"), configuration.Defaults.Enforce), resp) {
+		return
+	}
+
+	if reportError(validateVersion(field.NewPath("defaults", "enforce-version"), configuration.Defaults.EnforceVersion), resp) {
+		return
+	}
+	if reportError(validateLevel(field.NewPath("defaults", "warn"), configuration.Defaults.Warn), resp) {
+		return
+	}
+
+	if reportError(validateVersion(field.NewPath("defaults", "warn-version"), configuration.Defaults.WarnVersion), resp) {
+		return
+	}
+	if reportError(validateLevel(field.NewPath("defaults", "audit"), configuration.Defaults.Audit), resp) {
+		return
+	}
+
+	if reportError(validateVersion(field.NewPath("defaults", "audit-version"), configuration.Defaults.AuditVersion), resp) {
+		return
+	}
+
+	// validate exemptions
+	if reportError(validateUsernames(configurationTemplate), resp) {
+		return
+	}
+
+	if reportError(validateRuntimeClasses(configurationTemplate), resp) {
+		return
+	}
+
+	if reportError(validateNamespaces(configurationTemplate), resp) {
+		return
+	}
+}
+
+func reportDeletionError(clustersUsingTemplate int, err error, clusterType, templateName string, resp *admissionv1.AdmissionResponse) bool {
+	if err != nil {
+		resp.Result = &metav1.Status{
+			Status:  "Failure",
+			Message: fmt.Sprintf("failed to get %s clusters by index: %v", clusterType, err),
+			Reason:  metav1.StatusReasonInternalError,
+			Code:    http.StatusInternalServerError,
+		}
+		resp.Allowed = false
+		return true
+	}
+
+	if clustersUsingTemplate > 0 {
+		message := fmt.Sprintf("Cannot delete template '%s' as it is being used by %d %s clusters", templateName, clustersUsingTemplate, clusterType)
+		if clustersUsingTemplate == 1 {
+			message = fmt.Sprintf("Cannot delete template '%s' as it is being used by a %s cluster", templateName, clusterType)
```

Comment:
```suggestion
			message = fmt.Sprintf("Cannot delete template '%s' as it is being used by the '%s' cluster", templateName, clusterType)
```

Reply:
There can be many management and provisioning clusters, and because we don't include the name of the cluster I think `a` is the correct wording.